### ISSUE TITLE
Extend Tutorials Section

### ISSUE THREAD

**Issue by CursedRock17** [ros2/message_filters#130]
As per the title we could keep expanding the [tutorials section](https://github.com/ros2/message_filters/tree/rolling/doc/Tutorials) of our `message_filters` documentation because you can never have enough documentation. Specifically https://github.com/ros2/message_filters/pull/119#discussion_r1604565909 specifies for us to add an [ApproximateEpsilonTime](https://github.com/ros2/message_filters/blob/rolling/include/message_filters/sync_policies/approximate_epsilon_time.h) tutorials. That being said the following policies and objects could technically have their own tutorials:

- [ ] [ApproximateEpsilonTime](https://github.com/ros2/message_filters/blob/rolling/include/message_filters/sync_policies/approximate_epsilon_time.h) (C++)
- [ ] [LatestTime](https://github.com/ros2/message_filters/blob/ac26507686ea8533504bb917128659951029759d/include/message_filters/sync_policies/latest_time.h) (C++)
- [ ] [Chain](https://github.com/ros2/message_filters/blob/ac26507686ea8533504bb917128659951029759d/include/message_filters/chain.h#L104) (C++)
- [x] [Cache](https://github.com/ros2/message_filters/blob/ac26507686ea8533504bb917128659951029759d/include/message_filters/cache.h#L62) (C++)
- [x] [Cache](https://github.com/ros2/message_filters/blob/ac26507686ea8533504bb917128659951029759d/src/message_filters/__init__.py#L93) (Python)

Possibly:

- [ ] [Custom Simple Filter](https://github.com/ros2/message_filters/blob/ac26507686ea8533504bb917128659951029759d/include/message_filters/simple_filter.h#L51) (C++ - Intermediate)
- [ ] [Custom Simple Filter](https://github.com/ros2/message_filters/blob/ac26507686ea8533504bb917128659951029759d/src/message_filters/__init__.py#L46) (Python - Intermediate)

I should be able to get to most of these relatively fast, but they should be in one location ready for tracking. Additionally since most of tfhe documentation is hosted on `message_filters` this shouldn't need to go to `ros2_documentation`

**Comment by EsipovPA** [comment]
Is someone working on this one? If nobody is, is it ok, If I try to add at least some of them?

Looking for some issues to work on in some spare time. 🙂 

upd

I've started with the `Cache` filter for `Python`, as I've worked with lately. If anybody is OK with it.

**Comment by CursedRock17** [comment]
Hey @EsipovPA, I have not been able to get to this, maybe the issue will end up being assigned to you. The more the merrier.

**Comment by EsipovPA** [comment]
Added the PR for Python Cache filter. Start working on C++ tutorial for the same filter.

upd

To add tutorial with String for Cache filter in Cpp, it will require the Cache filter to caching headerless messages. This change is suggested in this issue: #74. I'll try to tackle it in the PR with the tutorial itself.

**Comment by EsipovPA** [comment]
Seems like there is an issue with ["Build your package"](https://docs.ros.org/en/rolling/p/message_filters/doc/Tutorials/Cache-Python.html#build-your-package) section in tutorial for Cache (Python). Not sure about the possible solutions.

**Comment by EsipovPA** [comment]
I thik this issue is not closed yet. I've added only one tutorial and working on them one by one. Is it possible to reopen this issue? @CursedRock17, @ahcorde

**Comment by CursedRock17** [comment]
@EsipovPA i would agree there's still lots to do, I did mark off the Cache filter in the tasks list though.

**Comment by EsipovPA** [comment]
[Added PR](https://github.com/ros2/message_filters/pull/196) with the tutorial for `C++` for `Cache`. Reviews are welcome! `:)`

**Comment by EsipovPA** [comment]
@CursedRock17. I think we may check the [Cache](https://github.com/ros2/message_filters/blob/ac26507686ea8533504bb917128659951029759d/include/message_filters/cache.h#L62) (C++) checkmark. Started working on `Chain` filter tutorial. BTW, shoudn't we add the same filter for `Python`? Did not look at the code yet, sorry if this is a stupid question.

**Comment by CursedRock17** [comment]
@EsipovPA awesome! As for why there's no `Chain` class in Python is beyond me, it looks like that C file was last changed 7 years ago and it's probably just overlooked since. If you haven't started on it, I could work on the class implementaiton + tutorial all in Python some time this weekend.

**Comment by EsipovPA** [comment]
Hey everyone!

I was out of town for the last couple of weeks with my family, and will be out of town again for the two weeks, starting the 4th of august.

After that, will get back to work on this issue.

**Comment by EsipovPA** [comment]
Hello everyone! I've added a [PR](https://github.com/ros2/message_filters/pull/203) with a new tutorial for the ``Chain`` filter. I would be great if anyone could review it!

**Comment by EsipovPA** [comment]
I think I might turn to adding a Python implementation for the `Chain` filter, as it is missing. And with that I'll try to add a tutorial for that as well.

**Comment by EsipovPA** [comment]
I've also noticed that there is some problem with `Build your package` section in the `Cache` tutorial both for C++ and Python. But when I build documentation locally, I do not see this problem. Maybe the environment where the documentation is build for web-page lacks `python3-sphinx-tabs` package? I've had some issues with the absence of that specific package, when I was building the documentation locally for the first time.

<img width="1688" height="483" alt="Image" src="https://github.com/user-attachments/assets/70149756-9e34-4de5-bab9-92138a8f95ba" />

The left part of the image is my local build and the right one is from the web-page

**Comment by EsipovPA** [comment]
I think this issue is not completely closed. Maybe reopen it again? And add a checkmark for Chain (C++)?

**Comment by ahcorde** [comment]
It was closed automatically, reopen again

**Comment by EsipovPA** [comment]
Opened a [PR](https://github.com/ros2/message_filters/pull/219) with `Chain` tutorial for for `Python`. Started working on custom `SimpleFilter`.

**Comment by EsipovPA** [comment]
It took slightly more time that I've expected to do the `Custom SimpleFilter` tutorials. I've completed the tutorial for `Python`. I'll open a `PR` during the course of this week. I need to read it out for errors, typos and stuff like that. Sorry for the delay.

**Comment by EsipovPA** [comment]
Opened a PR with SimpleFilter inheritance tutorial. Will be greatfull for reviews! `:)`

**Comment by EsipovPA** [comment]
I think we may check the 'Simple filter for python' checkmark. `:)`

upd

Started working on `SimpleFilter for CPP`

**Comment by EsipovPA** [comment]
Opened a `PR` with `SimpleFilter` inheritance tutorial for `C++`. Will be grateful for reviews!

**Comment by kscottz** [comment]
@EsipovPA I am going to assign this issue to you to get it off of the "todo" queue. If that's a problem please unassign yourself.

**Comment by EsipovPA** [comment]
@kscottz Hello! Not a problem at all! I was planning to complete this issue eventually.

**Comment by EsipovPA** [comment]
Started working on tutorial for `ApproximateEpsilonTime`. Should not there be also a python implementation? I may add one, if it sounds reasonable, but later.

**Comment by EsipovPA** [comment]
@CursedRock17 Hi! We've merged a C++ tutorial for [ApproximateEpsilonTime](https://github.com/ros2/message_filters/blob/rolling/include/message_filters/sync_policies/approximate_epsilon_time.h). I think we may check the corresponding checkmark. I've started working on the [LatestTime](https://github.com/ros2/message_filters/blob/ac26507686ea8533504bb917128659951029759d/include/message_filters/sync_policies/latest_time.h) tutorial. Hope to open a PR within a week (If I'm begin lucky with some free time for that)

**Comment by EsipovPA** [comment]
Yay! [Opened a PR](https://github.com/ros2/message_filters/pull/266) for the `LatestTime` policy!
